vm/systemSmartContracts/validator.go

```diff
@@ -1127,12 +1160,19 @@ func (v *validatorSC) activateStakingFor(
 		}
 
 		if stakedData.UnStakedNonce == 0 {
-			numRegistered++
+			numActivatedKey++
+		}
+
+		nodesActivated++
+		if nodesActivated >= maxNumNodesToActivate {
+			break
 		}
 	}
 
-	registrationData.NumRegistered = uint32(numRegistered)
-	registrationData.LockedStake.Mul(fixedStakeValue, big.NewInt(0).SetUint64(numRegistered))
+	registrationData.NumRegistered = uint32(numActivatedKey)
+	registrationData.LockedStake.Mul(fixedStakeValue, big.NewInt(0).SetUint64(numActivatedKey))
```

Comment:
If you increase `nodesActivated` two times above, is this still backwards compatbile?

Reply:
those are 2 different variables.